diff -uN --exclude=.svn --exclude=PagePrint-bin --exclude=dead --exclude=.libs ./HttpObserver.cpp ../../../pageprint-dev/trunk/src/HttpObserver.cpp
--- ./HttpObserver.cpp	2006-09-18 15:37:26.000000000 +0200
+++ ../../../pageprint-dev/trunk/src/HttpObserver.cpp	2006-09-18 15:44:14.000000000 +0200
@@ -7,8 +7,9 @@
 #include "Config.h"
 #include <glib.h>
 #include "nsIHttpChannel.h"
-#include "nsXPIDLString.h"
+#include "nsStringAPI.h"
 #include "nsIServiceManager.h"
+#include "nsServiceManagerUtils.h"
 #include "nsIObserverService.h"
 
 HttpObserver::HttpObserver(void)
diff -uN --exclude=.svn --exclude=PagePrint-bin --exclude=dead --exclude=.libs ./Makefile ../../../pageprint-dev/trunk/src/Makefile
--- ./Makefile	2006-09-18 15:37:26.000000000 +0200
+++ ../../../pageprint-dev/trunk/src/Makefile	2006-09-18 15:34:19.000000000 +0200
@@ -1,6 +1,6 @@
 BASEDIR = .
 
-INCLUDES = `pkg-config --cflags mozilla-gtkmozembed gtk+-2.0 mozilla-xpcom libxml-2.0` -I/usr/include/mozilla-1.7.13/gfx/
+INCLUDES = `pkg-config --cflags mozilla-gtkmozembed gtk+-2.0 mozilla-xpcom libxml-2.0` -I/usr/include/mozilla-seamonkey-1.0.5/gfx/
 
 LIBS = `pkg-config --libs mozilla-gtkmozembed mozilla-xpcom gtk+-2.0 libxml-2.0`
 
diff -uN --exclude=.svn --exclude=PagePrint-bin --exclude=dead --exclude=.libs ./PagePrint ../../../pageprint-dev/trunk/src/PagePrint
--- ./PagePrint	2006-09-18 15:37:26.000000000 +0200
+++ ../../../pageprint-dev/trunk/src/PagePrint	2006-09-18 15:48:25.000000000 +0200
@@ -5,7 +5,7 @@
 
 // fine mozilla dir
 $mozDir = "/usr/lib/mozilla";
-$dirs = glob("/usr/lib/mozilla*");
+$dirs = glob("/usr/lib/mozilla-seamonkey*");
 usort($dirs,'mozSort');
 $mozDir = array_pop($dirs);
 
@@ -13,7 +13,7 @@
 	return version_compare(getVer($a),getVer($b));
 }
 function getVer($in) {
-	if (preg_match('/mozilla-(.*)/',$in,$match)) {
+	if (preg_match('/mozilla-seamonkey-(.*)/',$in,$match)) {
 		return $match[1];
 	}
 	return 0;
diff -uN --exclude=.svn --exclude=PagePrint-bin --exclude=dead --exclude=.libs ./PagePrint.cpp ../../../pageprint-dev/trunk/src/PagePrint.cpp
--- ./PagePrint.cpp	2006-09-18 15:37:26.000000000 +0200
+++ ../../../pageprint-dev/trunk/src/PagePrint.cpp	2006-09-18 15:46:45.000000000 +0200
@@ -9,13 +9,12 @@
 #include <glib.h>
 #include <glib/gprintf.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
-//#include <glib/gstdio.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
 #include <fcntl.h>
-
+#include <pwd.h>
 
 // mozilla headers
 #include "prenv.h"
@@ -24,12 +23,11 @@
 #include "nsXPCOM.h"
 
 // Mozilla Printing Stuff
-#include "nsIPrintOptions.h"
 #include "nsIPrintSettings.h"
 #include "nsIWebBrowserPrint.h"
 #include "nsCOMPtr.h"
 #include "nsIInterfaceRequestorUtils.h"
-#include "nsString.h"
+#include "nsEmbedString.h"
 
 // print progress code
 #include "PrintProgress.h"
@@ -177,18 +175,18 @@
 		exit(1);
 	}
 
-	nsAutoString mPrintToFile, mPrinter, mPaperName;
+	nsEmbedString mPrintToFile, mPaperName;
 
 	gchar* printFile = g_strdup_printf("/tmp/PrintPage%d.ps",getpid());
 
-	mPrintToFile.AssignWithConversion(printFile);
-	mPaperName.AssignWithConversion(ppConf->printPaperType);
+	NS_CStringToUTF16(nsEmbedCString(printFile), NS_CSTRING_ENCODING_UTF8, mPrintToFile);
+	NS_CStringToUTF16(nsEmbedCString(ppConf->printPaperType), NS_CSTRING_ENCODING_UTF8, mPaperName);
 
-	settings->SetToFileName (ToNewUnicode(mPrintToFile));
+	settings->SetToFileName (mPrintToFile.get());
 
 	settings->SetPrintToFile (PR_TRUE);
 
-	settings->SetPaperName (ToNewUnicode(mPaperName));
+	settings->SetPaperName (mPaperName.get());
 
 	settings->SetPrintSilent (PR_TRUE);
 
@@ -210,12 +208,13 @@
 	settings->SetMarginLeft(ppConf->printMarginLeft);
 	settings->SetMarginRight(ppConf->printMarginRight);
 
-	nsAutoString empty;
-	empty.AssignWithConversion(" ");
-	settings->SetFooterStrLeft(ToNewUnicode(empty));
-	settings->SetFooterStrRight(ToNewUnicode(empty));
-	settings->SetHeaderStrLeft(ToNewUnicode(empty));
-	settings->SetHeaderStrRight(ToNewUnicode(empty));
+	nsEmbedString empty;
+	NS_CStringToUTF16(nsEmbedCString(" "), NS_CSTRING_ENCODING_UTF8, empty);
+	
+	settings->SetFooterStrLeft(empty.get());
+	settings->SetFooterStrRight(empty.get());
+	settings->SetHeaderStrLeft(empty.get());
+	settings->SetHeaderStrRight(empty.get());
 
 	PrintProgress *listener = new PrintProgress(printFile,ppConf->outputFile);
 	result = print->Print(settings, listener);
@@ -344,16 +343,16 @@
 	gtk_init(&argc, &argv);
 
 	// setup mozilla environment
-	char *home_path;
+	struct passwd *pass;
 	char *full_path;
-	home_path = PR_GetEnv("HOME");
-	if (!home_path) {
-		fprintf(stderr, "Failed to get HOME\n");
-		g_print("Failed to get HOME\n");
+	
+	if (pass = getpwuid(getuid())) {
+		full_path = g_strdup_printf("%s/%s", pass->pw_dir, ".PagePrint");
+	} else {
+		g_print("getpwuid failed!");
 		return 1;
 	}
-
-	full_path = g_strdup_printf("%s/%s", home_path, ".PagePrint");
+	
 	gtk_moz_embed_set_profile_path(full_path, "PagePrint");
 	g_print("Profile path is: %s\n",full_path);
 
